equistore-core/src/io/npy_header.rs

```diff
@@ -661,5 +664,9 @@ mod tests {
             shape: vec![3],
         };
         assert_eq!(header.to_dict_literal(), "{ 'descr': '<f8', 'fortran_order': True, 'shape': (3, ) }");
+
+        // check round trip
+        let parsed = Header::from_str(&header.to_dict_literal()).unwrap();
```

Comment:
@PicoCentauri this is the test checking that we can read again the header we produced, which is the root cause of the bug

Reply:
ahh okay then I am happy and we can merge!